compiler/AST/AstDump.cpp

```diff
@@ -81,8 +84,25 @@ void AstDump::view(const char* passName, int passNum) {
       AstDump logger;
 
       if (logger.open(module, passName, passNum) == true) {
-        module->accept(&logger);
-        logger.close();
+        if(fLogFormat == LogFormat::NPRINT) {
+          // This is an absolute hack. nprint_view prints directly to stdout, we should
+          // update it to take a file pointer use it and pass it along as needed. But rather
+          // than update all the code I instead redirect stdout to the file (and then restore it
```

Comment:
Can we not make this happen by adding an optional argument to `nprint_view`?

Reply:
If a quick try doesn't deliver results, you can merge the PR as-is. Ideally, a private issue could be nice as a rainy-day task for the future.

Reply:
> Can we not make this happen by adding an optional argument to nprint_view?

You mean passing in a FILE* or id as an optional arg (and have it default to stdout)? Yes I agree, that's what we would want to do. We'd then have to update all calls to `printf` to `fprintf` or whatever.

I actually don't think it'd be that bad, could be done in less-than a day, but it's more than a few minutes of work. I agree would be a nice rainy-day task. I'll create an issue to follow up here: https://github.com/Cray/chapel-private/issues/5608

Reply:
Oh, I thought you were shying away from updating all the calls to `nprint_view` itself, and that got me thinking "why not an optional argument?". Ok that makes sense.